group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -3071,6 +3072,41 @@ private void removeCurrentMemberFromGenericGroup(
         group.remove(member.memberId());
     }
 
+    public CoordinatorResult<DeleteGroupsResponseData.DeletableGroupResultCollection, Record> groupDelete(
+            RequestContext context,
+            List<String> groupIds
+    ) throws ApiException {
+        final DeleteGroupsResponseData.DeletableGroupResultCollection resultCollection =
+                new DeleteGroupsResponseData.DeletableGroupResultCollection();
+        final List<Record> records = new ArrayList<>();
+
+        groupIds.forEach(groupId -> {
+            DeleteGroupsResponseData.DeletableGroupResult result =
+                    new DeleteGroupsResponseData.DeletableGroupResult().setGroupId(groupId);
+            try {
+                validateGroupDelete(groupId);
+                records.add(RecordHelpers.newGroupMetadataTombstoneRecord(groupId));
```

Comment:
`newGroupMetadataTombstoneRecord` only works for generic groups. For consumer groups, we need to write other tombstones.

Reply:
done